### PR TITLE
[Harvester] Fetch the current Trust Bundle from a SPIRE Server

### PR DESCRIPTION
**Pull request by mchurichi**
Signed-off-by: Maximiliano Churichi <mchurichi@gmail.com>

**Pull Request check list**

- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**
N/A

**Description of change**
Implements the SPIRE plugin interface to get a SPIRE Server bundle via UDS.

Test steps:
- Run a SPIRE Server
- Change the default config file of the harvester and set `spire_socket_path` to the UDS of the SPIRE Server
- Run `go run cmd/harvester/main.go spire-bundle`

**Which issue this PR fixes**
N/A

### CODE REVIEW

File: cmd/harvester/cli/cli.go [hunk 1]
```diff
@@ -2,11 +2,13 @@ package cli
```
Comment (mchurichi) [inline cmd/harvester/cli/cli.go:2]: The changes introduced into this file are only intended to facilitate the testing of this feature and will be removed before merging this PR

File: pkg/harvester/spire/bundleclient_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		clientBundle  *types.Bundle
+		protoParseErr string
+	}{
+		{
+			name:         "ok",
+			expected:     spiffebundle.New(spiffeid.RequireTrustDomainFromString("example.org")),
+			clientBundle: &types.Bundle{TrustDomain: "example.org"},
+		},
+		{
+			name:      "error_calling_client",
```
Comment (jufantozzi) [inline pkg/harvester/spire/bundleclient_test.go:37]: just out of curiosity: any reason we're using camel case for these names instead of regular spaces?

File: pkg/harvester/spire/client_fake_test.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
+package spire
```
Comment (jufantozzi) [inline pkg/harvester/spire/client_fake_test.go:1]: Could it be considered anti-patter to have files ending with `_test.go` that are not actually running tests? Haven't really thought about this yet...

**Review by jufantozzi — APPROVED** [review]
Left some questions only 💯 🎖️

File: cmd/harvester/cli/cli.go [hunk 4]
```diff
@@ -22,22 +24,37 @@ func NewHarvesterCLI() *HarvesterCLI {
...
+// TODO: the following methods are only intended for testing and should be replaced by a proper CLI implementation
+func (c *HarvesterCLI) spireBundle(ctx context.Context, cfg *config.HarvesterConfig) int {
+	server := spire.NewLocalSpireServer(cfg.HarvesterConfigSection.SpireSocketPath)
+	bundle, err := server.GetBundle(ctx)
+	if err != nil {
+		c.logger.Error("Error getting bundle:", err)
+		return 1
+	}
+	bytes, _ := bundle.X509Bundle().Marshal()
+	c.logger.Debug(string(bytes))
```
[the remark this answers is not in the document]
Comment (mchurichi) [reply cmd/harvester/cli/cli.go:58]: Please ignore all the changes to this file as they will be reverted, they are only for testing purposes. That said, as the PEM bundle is the only output of this command I think this is enough for bundle display.

File: pkg/harvester/spire/bundleclient_test.go (hunk 2 above)
Comment (mchurichi) [reply pkg/harvester/spire/bundleclient_test.go:37]: oh no, just a habit 😄 we can change it if needed

**Review by maxlambrecht — APPROVED** [review]
🍰

File: pkg/harvester/spire/client_fake_test.go (hunk 3 above)
Comment (mchurichi) [reply pkg/harvester/spire/client_fake_test.go:1]: I don't know if it is an anti-pattern, but if we don't include `_test` some linters warn about unused code. Also, `go test` is not throwing any warning about this file, but it is warning about empty test files `testing: warning: no tests to run`.
Additionally,  I'd rather have all the code used for testing somehow separated to clearly identify that is not core code.

File: cmd/harvester/cli/spire.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
+package cli
```
Comment (mchurichi) [inline cmd/harvester/cli/spire.go:1]: Again, this file is only intended to facilitate testing via `go run cmd/harvester/main.go spire-bundle`. It will be removed before merging this PR

File: pkg/harvester/spire/client_fake_test.go (hunk 3 above)
Comment (jufantozzi) [reply pkg/harvester/spire/client_fake_test.go:1]: Ok, thanks for this context. I wouldn't mind about this warning, it does not seem like a problem to me.

> Additionally, I'd rather have all the code used for testing somehow separated to clearly identify that is not core code.

Agreed as well. As our testing infrastructure grows we'll have to figure that out. I wouldn't mind having it the way it is today, but it should see some changes as this gets more complex.

**Review by jufantozzi — APPROVED** [review]
💅'led it

File: pkg/harvester/spire/bundleclient_test.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
+package spire
```
Comment (maxlambrecht) [inline pkg/harvester/spire/bundleclient_test.go:1]: I think, as a general rule, tests should be placed in a different package (e.g.,`spire_test`) in order to enforce testing the exposed behavior and not internal details.

**Review by maxlambrecht — APPROVED** [review]
🍪

File: pkg/harvester/spire/bundleclient_test.go (hunk 6 above)
Comment (mchurichi) [reply pkg/harvester/spire/bundleclient_test.go:1]: We should discuss the pros and cons of this 💯

Comment (jufantozzi) [reply pkg/harvester/spire/bundleclient_test.go:1]: agreed as well!

**Review by jufantozzi — DISMISSED** [review]

**Comment by mchurichi** [comment]
@maxlambrecht @jufantozzi rebased and CLI removed

**Review by maxlambrecht — DISMISSED** [review]

**Review by jufantozzi — APPROVED** [review]
